### PR TITLE
DA-21/DA-57: Add visualization of object tracks

### PR DESCRIPTION
**Pull request by MarcBrau**
This PR adds the option to visualize object tracks in the ROS driver. The tracks are visualized as unfilled cubes (only the edges are visible) that represent the outlines of the current bounding box of this specific track. Further, each track has a velocity represented as an arrow starting in the center of the bounding box. The following image shows an example of a track bounding box (unfilled blue cube) together with the ego-vehicle (green cube).
image

### CODE REVIEW

File: src/tracks.cc [hunk 1]
```diff
@@ -0,0 +1,131 @@
+#include "zendar_ros_driver/tracks.h"
+namespace zen {
+visualization_msgs::MarkerArray
+Tracks(const zpb::drivable_area::Tracks& tracks){
+  visualization_msgs::MarkerArray track_msgs;
+  for (const zpb::drivable_area::Track& track : tracks.track()) {
+    // Iterate over edges of bounding box (= bounding box visualization
+    // represented by list of edges msgs)
+    for (std::size_t edge_id = 0; edge_id < track.bbox_edges().size() - 1;
```
Comment (hamlittle) [inline src/tracks.cc:9]: I don't know this well, but usually with a for loop I am used to seeing

```
edge_id < track.bbox_edges().size()
```

or

```
edge_id <= track.bbox_edges().size() - 1
```

Can you confirm this is correct?

```
edge_id < track.bbox_edges().size() - 1
```

File: src/tracks.cc [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  // Define header
+  edge_msg.header.frame_id = "map";
+  edge_msg.header.stamp = ros::Time::now();
+
+  edge_msg.ns = std::to_string(track.id());
+  edge_msg.id = edge_id;
+
+  // Define type of marker (0 = Arrow)
+  edge_msg.type = 0;
+  edge_msg.action = 0;
```
Comment (hamlittle) [inline src/tracks.cc:35]: are there #defines from ROS that you can use for these? or else make them yourself?

```
static constexpr int EDGE_TYPE_ARROW = 0
static constexpr int ACTION_TYPE_ARROW = 0
```

File: src/tracks.cc [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  pose.position.y = 0;
+  pose.position.z = 0;
+  pose.orientation.x = 0;
+  pose.orientation.y = 0;
+  pose.orientation.z = 0;
+  pose.orientation.w = 1;
+  edge_msg.pose = pose;
+
+  // Define scale (Only want shaft of arrow => Set head diameter/length to 0)
+  edge_msg.scale.x = 0.3;  // shaft diameter
```
Comment (hamlittle) [inline src/tracks.cc:51]: define a constant would be nice

File: src/tracks.cc [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+CreateVelocityMsg(const zpb::drivable_area::Track& track) {
+  visualization_msgs::Marker velocity_msg;
+
+  // Define header
+  velocity_msg.header.frame_id = "map";
+  velocity_msg.header.stamp =ros::Time::now();
+
+  // Define type of marker (0 = Arrow)
+  velocity_msg.type = 0;
+  velocity_msg.action = 0;
```
Comment (hamlittle) [inline src/tracks.cc:99]: CONSTAAAAAAAAAANTS for these please 🥺

**Review by hamlittle — CHANGES_REQUESTED** [review]
Logic looks good, but I left some comments about defining constants for the magic numbers in the code.

Comments are good, but constants that describe themselves are even better.

Can you please audit for those?

File: src/tracks.cc (hunk 2 above)
Comment (MarcBrau) [reply src/tracks.cc]: Thanks for pointing that out! Indeed there exist ROS constants for that. Addressed in 2977f8d.

File: src/tracks.cc (hunk 4 above)
Comment (MarcBrau) [reply src/tracks.cc]: Addressed in 2977f8d.

**Review by hamlittle — APPROVED** [review]
Looks good, thanks for addressing those.

Ship it.

**Pull request closed without merging**